contracts/proxy/utils/Initializable.sol

```diff
@@ -74,7 +99,29 @@ abstract contract Initializable {
         _;
     }
 
-    function _isConstructor() private view returns (bool) {
-        return !Address.isContract(address(this));
+    /**
+     * @dev Locks the contract, preventing any futur re-initialization. This cannot be part of an initializer call.
+     * Calling this in the constructor of an contract will prevent any direct initialization & re-initialization.
+     * It is recommended to use that to lock "implementation" contract that are designed to be called through proxies.
+     */
+    function _preventInitialize() internal virtual {
+        if (_setInitializedVersion(type(uint8).max)) {
+            _initializing = false;
+        }
```

Comment:
This code is simpler to understand the implications off, even though it's slightly more inefficient:

```suggestion
        _setInitializedVersion(type(uint8).max);
        _initializing = false;
```